sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OpenAiController.java

```diff
@@ -64,15 +64,10 @@ public static ResponseEntity<ResponseBodyEmitter> streamChatCompletionDeltas() {
     final Runnable consumeStream =
         () -> {
           final var totalOutput = new OpenAiChatCompletionOutput();
-          // try-with-resources ensures the stream is closed
-          try (stream) {
-            stream
-                .peek(totalOutput::addDelta)
-                .forEach(delta -> send(emitter, delta.getDeltaContent()));
-          } finally {
-            send(emitter, "\n\n-----Total Output-----\n\n" + objectToJson(totalOutput));
-            emitter.complete();
```

Comment:
Removing `emitter.complete();` breaks the endpoint. Also shouldn't the customer catch errors anyway?

Reply:
> Removing `emitter.complete();` breaks the endpoint. Also shouldn't the customer catch errors anyway?

Thanks, this change was not deliberate.

> Also shouldn't the customer catch errors anyway?

Reason for the change is to demonstrate the usage without `try`.
This is for demo / power-point purpose.
Even if we consider this not "best" practice.

So yes, in a perfect world customers catch errors. In a realistic world people forgot / don't care about try-catch on this level.

Reply:
Then don't forget to also update the other `streamChatCompletions` endpoint

Reply:
It's only now that I understood [Matthias concern above](https://github.com/SAP/ai-sdk-java/pull/49#discussion_r1754375596). Since I don't want to have the discussion, I will rather revert my changes :)